packages/ai/core/util/cosine-similarity.ts

```diff
@@ -1,23 +1,60 @@
+import { InvalidArgumentError } from '../../errors/invalid-argument-error';
+
 /**
  * Calculates the cosine similarity between two vectors. This is a useful metric for
  * comparing the similarity of two vectors such as embeddings.
  *
  * @param vector1 - The first vector.
  * @param vector2 - The second vector.
+ * @param options - Optional configuration.
+ * @param options.throwErrorForEmptyVectors - If true, throws an error for empty vectors. Default: false.
  *
  * @returns The cosine similarity between vector1 and vector2.
+ * @returns 0 if either vector is the zero vector.
+ * @throws {InvalidArgumentError} If throwErrorForEmptyVectors is true and vectors are empty.
  * @throws {Error} If the vectors do not have the same length.
  */
-export function cosineSimilarity(vector1: number[], vector2: number[]) {
+export function cosineSimilarity(
+  vector1: number[],
+  vector2: number[],
+  options?: {
+    throwErrorForEmptyVectors?: boolean;
+  },
+) {
+  // TODO: In the next major version, change the default value of throwErrorForEmptyVectors to true
+  const { throwErrorForEmptyVectors = false } = options ?? {};
+
+  if (throwErrorForEmptyVectors) {
```

Comment:
Considering the vector length check below, this can be simplified by pushing it after the check from line 44. After check from line 44, you only need to check vector1 length (bc vector2 length is the same). this will save a few bytes